### PR TITLE
Update syngap to v1.2.1

### PR DESCRIPTION
**Pull request by yanyew**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by yanyew** [comment]
@martin-g This time the PR failed in another wired way...
Is it somethind wrong about the Azure? I saw the same situation in other PR like https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=58625&view=logs&jobId=eb1ca712-fec5-5d93-deb0-40983fe61f2f&j=eb1ca712-fec5-5d93-deb0-40983fe61f2f
image

**Comment by martin-g** [comment]
This is a known issue with the Azure CI - https://github.com/bioconda/bioconda-recipes/issues/47856
All PRs are affected!
We have to wait until it is resolved by Microsoft.

**Comment by yanyew** [comment]
@martin-g Hello, this PR somehow worked. Can you review it? Thanks!

File: recipes/syngap/build.sh [hunk 1]
```diff
@@ -1,12 +1,21 @@
+#!/usr/bin/env bash
+set -xe
+df -h
+du -sch $CONDA_PREFIX
```
Comment (martin-g) [inline recipes/syngap/build.sh:4]: The above two lines are no more needed. They were added to debug the issue.

File: recipes/syngap/build.sh [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
 mkdir -p $PREFIX/bin
 cp -rf $SRC_DIR/bin $PREFIX/bin
 cp -rf $SRC_DIR/scripts $PREFIX/bin
+df -h
```
Comment (martin-g) [inline recipes/syngap/build.sh:8]: Same.
And few more below.

File: recipes/syngap/meta.yaml [hunk 3]
```diff
@@ -1,38 +1,24 @@
...
-    - numpy
-    - pandas
-    - matplotlib-base
+  run_exports:  '{{ pin_subpackage(name, max_pin="x") }}'
 
 requirements:
-  bulid: 
-    - python >=3.10
+  #bulid: 
+  #  - python >=3.10
```
Comment (martin-g) [inline recipes/syngap/meta.yaml:21]: Please remove the commented out lines.

File: recipes/syngap/meta.yaml [hunk 4]
```diff
@@ -83,7 +69,9 @@ test:
     - genepair
     - evi
     - eviplot
-  commands: 
+  commands:
+    - df -h
+    - du -sch $CONDA_PREFIX
```
Comment (martin-g) [inline recipes/syngap/meta.yaml:74]: Remove the debug

File: recipes/syngap/meta.yaml [hunk 5]
```diff
@@ -93,6 +81,7 @@ test:
     - syngap genepair -h
     - syngap evi -h
     - syngap eviplot -h
+    - df -h
```
Comment (martin-g) [inline recipes/syngap/meta.yaml:84]: ditto

**Review by yanyew — COMMENTED** [review]
All mentioned modification suggestions have been completed

File: recipes/syngap/meta.yaml (hunk 5 above)
Comment (yanyew) [reply recipes/syngap/meta.yaml]: > ditto

It's so wired that when I followed your suggestions, the the Azure CI met the error like below https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=61168&view=logs&s=5162d36d-59e0-5d46-757c-443386a65b88&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f
image
image

Comment (yanyew) [reply recipes/syngap/meta.yaml]: And here is the previous successful run before the modification, which didn't meet the "no space left on device" error.
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=61117&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=4dddc55b-10bc-50fc-ac68-7ff899b7a030

**Pull request closed without merging**